frontend/app/src/composables/history/events/events-query-status.ts

```diff
@@ -0,0 +1,61 @@
+import { type ComputedRef } from 'vue';
+import { type MaybeRef } from '@vueuse/core';
+import { type HistoryEventsQueryData } from '@/types/websocket-messages';
+
+export const useEventsQueryStatus = (locations: MaybeRef<string[]> = []) => {
```

Comment:
Can you extract a common core with this and the tx-query-status they have similarities